src/app/base/components/AppSideNavigation/AppSideNavigation.tsx

```diff
@@ -94,6 +104,24 @@ const AppSideNavigation = (): JSX.Element => {
   });
   const themeColor = theme ? theme : maasTheme ? maasTheme : "default";
 
+  const filteredGroups = useMemo(() => {
+    if (hasVirsh === false && kvmsLoaded) {
+      const kvmGroupIndex = navGroups.findIndex(
+        (group) => group.groupTitle === "KVM"
+      );
+
+      const virshItemIndex = navGroups[kvmGroupIndex].navLinks.findIndex(
+        (navLink) => navLink.label === "Virsh"
+      );
+
+      if (virshItemIndex > -1) {
+        navGroups[kvmGroupIndex].navLinks.splice(virshItemIndex, 1);
+      }
+    }
+
+    return navGroups;
+  }, [hasVirsh]);
```

Comment:
```suggestion
  }, [hideVirsh]);
```

Reply:
Ideally we should use something like an array of hidden links instead to make it easier to extend in the future, but I'm not gonna push for it. It's ok as is.